docs: update docs for the extends option

burhanuday opened the pull request:
_describe your changes..._

Adds docs for the `extends` option added to webpack-cli in PR https://github.com/webpack/webpack-cli/pull/3738

- [x] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [x] Make sure your PR complies with the [writer's guide][2].
- [x] Review the diff carefully as sometimes this can reveal issues.
- [x] Do not abandon your Pull Request: [Stale Pull Requests][3].

Closes https://github.com/webpack/webpack.js.org/issues/6801

burhanuday commented:
PR not to be merged until a new release is done for webpack-cli

snitin315 reviewed (commented):
Thanks for the PR. I'll review this by today/tomorrow.

burhanuday commented:
One thing is left to be added. This feature will be supported after the release of webpack and webpack-cli
So as a warning, we need to add the minimum webpack and webpack-cli versions required to use this feature

snitin315 commented on src/content/configuration/extending-configurations.mdx:
```diff
@@ -0,0 +1,122 @@
+
+`string | string[]`
+
+<Badge text="webpack v5.82.0+" />
+<Badge text="webpack-cli v5.0.3+" />
```
How does this look?

```suggestion
<Badge text="webpack v5.82.0+" /> <Badge text="webpack-cli v5.0.3+" />
```

burhanuday replied:
the same basically
<img width="827" alt="image" src="https://user-images.githubusercontent.com/25934701/236656587-c74687d4-b28c-4c46-a084-1887a5867dd0.png">

snitin315 commented on src/content/configuration/extending-configurations.mdx:
```diff
@@ -0,0 +1,122 @@
+
+<Badge text="webpack v5.82.0+" />
+<Badge text="webpack-cli v5.0.3+" />
+
+The `extends` property allows you to extend an existing configuration to use as base. It internally uses the `webpack-merge` package to merge the configurations and helps you to avoid duplicating configuration between multiple configurations.
+
+W> **This option is not supported via the Node API**: Extends will have no effect when using the Node API. webpack-cli is required to use this feature.
```
```suggestion

The `extends` property allows you to extend an existing configuration to use as the base. It internally uses the `webpack-merge` package to merge the configurations and helps you to avoid duplicating configurations between multiple configurations.

W> **This option is not supported via the Node API**: Extends will have no effect when using the Node API. webpack-cli is required to use this feature.
```

snitin315 commented on src/content/configuration/extending-configurations.mdx:
````diff
@@ -0,0 +1,122 @@
+
+```javascript
+module.exports = {
+  extends: './base.webpack.config.js',
````
```suggestion
  extends: path.resolve(__dirname, './base.webpack.config.js'),
```

snitin315 commented on src/content/configuration/extending-configurations.mdx:
```diff
@@ -0,0 +1,122 @@
+
+You can extend multiple configurations at once by passing an array of configuration paths to the `extends` property.
+
+Configurations from the `extends` property are merged from right to left, meaning that the configuration on the right will be merged into the configuration on the left. Configuration can be overriden by passing the same property in the configuration on the right.
```
```suggestion
Configurations from the `extends` property are merged from right to left, meaning that the configuration on the right will be merged into the configuration on the left. Configuration can be overridden by passing the same property in the configuration on the right.
```

snitin315 commented on src/content/configuration/extending-configurations.mdx:
````diff
@@ -0,0 +1,122 @@
+
+```javascript
+module.exports = {
+  extends: ['./js.webpack.config.js', './css.webpack.config.js'],
````
```suggestion
  extends: [path.resolve(__dirname, './js.webpack.config.js'), path.resolve(__dirname, './css.webpack.config.js')],
```

snitin315 commented on src/content/configuration/extending-configurations.mdx:
```diff
@@ -0,0 +1,122 @@
+
+## Loading configuration from external packages
+
+You can also load configuration from third party packages by passing the package name to the `extends` property. The package must export the webpack configuration in package.json.
```
```suggestion
You can also load configuration from third-party packages by passing the package name to the `extends` property. The package must export the webpack configuration in package.json.
```

snitin315 commented on src/content/configuration/extending-configurations.mdx:
````diff
@@ -0,0 +1,122 @@
+
+```javascript
+module.exports = {
+  extends: require.resolve('webpack-config-foo'),
````
I think we need better documentation here on how to create and publish `webpack-config-foo`. But we can add it later.

burhanuday replied:
> I think we need better documentation here on how to create and publish `webpack-config-foo`. But we can add it later.

More work needs to be done here. Ideally, consumers of a config should be able to pass options like we do for plugins.

burhanuday replied:
I'll create a new issue for this on the webpack-cli project

snitin315 reviewed (commented):
Looks good overall, added a few suggestions. Would be great to add an example that shows how we can override a config from the base config.

burhanuday commented:
> Looks good overall, added a few suggestions. Would be great to add an example that shows how we can override a config from the base config.

Added a section for this

snitin315 commented on src/content/api/cli.mdx:
````diff
@@ -540,6 +542,14 @@ You can merge two or more different webpack configurations with the help of `--m
 npx webpack --config ./first.js --config ./second.js --merge
 ```
 
+### extends
+
+You can extend existing webpack configurations with the help of `--extends`:
````
```suggestion
### extends

<Badge text="webpack-cli v5.1.0+" />

You can extend existing webpack configurations with the help of `--extends`:
```

snitin315 reviewed (approved):
Looks good. Thanks for contributing on this 🚀
